### DISCUSSION TITLE
Delete Button is working but at the wrong end

### DISCUSSION THREAD

**Discussion by Charlotte-Ketteler** [neuefische/questions#678] [Web] [answered]
Hi everyone, in my booking app I am trying to delete my Card in my BookingOverview/BookingOverviewPage. I wrote my handleDelete function into my app.js as I need my timeSlot State. My console.log is reading the id, so that works. Unfortunately my delete button is deleting my timeSlots Buttons in my BookingPage, which of course I don't want. I can't quite figure out where the mistake is. Maybe you can help. :) [Branch: Terminübersicht](https://github.com/Charlotte-Ketteler/capstone-project/tree/Termin%C3%BCbersicht)

**Comment by Kheirah** [comment, marked as answer]
Your delete function is passed down properly, very good!

The issue is that you're filtering the timeslots _instead of the bookings_ within the timeslots. And as it stands right now, you can't delete your bookings because they don't have ids.

In total, what you need to do to be able to delete bookings:
- manually add ids in all your bookings in `lib/data.js`:
```js
{
    id: 1,
    startTime: "08:00",
    endTime: "08:50",
    bookings: [
      {
        id: 1, //add some unique values for the id
        numberOfPeople: 1,
        horses: [
          {
            horseId: 1,
          },
...
```

- change the `handleSubmit` function by adding an id for every new booking in  `_app.js`:
```js
function handleSubmit(booking) {
    const timeSlotToBeUpdated = timeSlots.find((slot) => slot.id === booking.id);
    if (timeSlotToBeUpdated) {
      const horsesBooked = booking.horses.map((horseID) => ({ horseId: horseID }));
      const bookingID = Math.random().toString(32).substring(2);
      const updatedTimeSlot = { ...timeSlotToBeUpdated, bookings: [...timeSlotToBeUpdated.bookings, { id: bookingID, numberOfPeople: booking.numberOfPeople, horses: horsesBooked }] }
      setTimeSlots(timeSlots.map((slot) => slot.id === booking.id ? updatedTimeSlot : slot));
      setRecentlyBooked({ ...booking, bookingID });
      router.push("/BookingSuccessful");
    }
  }
```
- change the `delete` function:
```js
function handleDelete(id, bookingID) {
    setTimeSlots(timeSlots.map((slot) => slot.id === id ? { ...slot, bookings: slot.bookings.filter((booking) => booking.id !== bookingID) } : slot));
    setRecentlyBooked();
    router.push("/");
  }
```
- and finally in the `BookingOverview` component, add the `bookingID`:
```js
<button type="button" onClick={() => handleDelete(recentlyBooked.id, recentlyBooked.bookingID)}>löschen</button>
```
This should do it. But then your `BookingForm` needs further adjusting because it doesn't correctly display bookings. Something for you to look into =)

**Reply by Charlotte-Ketteler** [reply]
Thank you so much! :)
